#! /bin/sh -e
## 16_fix_opengl_check.dpatch by Martin Loschwitz <madkiss@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix handling of libGL.so.1 to not require the -dev packages (kudos to Ben Burton)

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

diff -ruN qt-x11-free-3.3.3.old/src/opengl/qgl_x11.cpp qt-x11-free-3.3.3/src/opengl/qgl_x11.cpp
--- qt-x11-free-3.3.3.old/src/opengl/qgl_x11.cpp	2004-08-05 14:42:12.000000000 +0000
+++ qt-x11-free-3.3.3/src/opengl/qgl_x11.cpp	2004-08-14 12:30:59.000000000 +0000
@@ -125,7 +125,7 @@
     if (gl_syms_resolved)
 	return TRUE;
 
-    QLibrary gl("GL");
+    QLibrary gl("/usr/lib/libGL.so.1");
     gl.setAutoUnload(FALSE);
 
     qt_glCallLists = (_glCallLists) gl.resolve("glCallLists");
